back/boxtribute_server/models/crud.py

```diff
@@ -9,33 +9,49 @@
 from ..exceptions import BoxCreationFailed, RequestedResourceNotFound
 from .definitions.beneficiary import Beneficiary
 from .definitions.box import Box
+from .definitions.location import Location
 from .definitions.qr_code import QrCode
 from .definitions.x_beneficiary_language import XBeneficiaryLanguage
 from .utils import utcnow
 
 BOX_LABEL_IDENTIFIER_GENERATION_ATTEMPTS = 10
 
 
-def create_box(data):
+def create_box(
+    product_id,
+    location_id,
+    created_by_id,
+    size_id,
+    comment="",
+    items=0,
+    qr_code_code=None,
+):
     """Insert information for a new Box in the database. Use current datetime
     and box state "InStock" by default. Generate an 8-digit sequence to identify the
     box. If the sequence is not unique, repeat the generation several times. If
     generation still fails, raise a BoxCreationFailed exception.
     """
+
     now = utcnow()
-    qr_code = data.pop("qr_code", None)
-    qr_id = QrCode.get_id_from_code(qr_code) if qr_code is not None else None
+    qr_id = QrCode.get_id_from_code(qr_code_code) if qr_code_code is not None else None
 
+    location_box_state = Location.get(Location.id == location_id).box_state
```

Comment:
You can use the convenience method `Location.get_by_id(location_id)`. By using `box_state_id` you can avoid that peewee performs a full `SELECT` of the BoxState FK model, instead only the integer ID is returned (the docs hide it in [this section](http://docs.peewee-orm.com/en/latest/peewee/models.html#foreignkeyfield); noted to myself that I add this to the Readme about peewee quirks :slightly_smiling_face: )